src/Microsoft.OData.Core/JsonLight/ODataJsonLightBatchPayloadItemPropertiesCache.cs

```diff
@@ -198,7 +241,7 @@ private void ScanJsonProperties()
 
                         case PropertyNameDependsOn:
                             {
-                                IList<string> dependsOnIds = new List<string>();
+                                List<string> dependsOnIds = new List<string>();

```

Comment:
@KenitoInc Because of the investigation previously done by @habbes...
`IList<T>` will use more memory than `List<T>`, because `List<T>` will use `public Enumerator GetEnumerator(){}` which returns an unboxed struct.
Whenever the interface versions of `GetEnumerator()` are called, the struct is boxed and returned as an interface type (`IEnumerator`). This causes an object to be allocated on the heap.
Plus there's actually no real value in using `IList` in the local variable here

Because